File: API/Helpers/Summary.cs
using API.Entities;

namespace API.Helpers;

public class Summary
{
    public Employee Employee { get; set; } = null!;
    public double TotalWorkShiftMinutes { get; set; }
    public double TotalWorkRecordMinutes { get; set; }
    public IEnumerable<SummaryWorkDetail> SummaryWorkDetails { get; set; } = [];
}

File: client/src/app/_models/summary.ts
import { Employee } from "./employee";
import { WorkShift } from "./workShift";

export interface Summary{
    employee: Employee;
    totalWorkShiftMinutes: number;
    totalWorkRecordMinutes: number;
    summaryWorkDetails: SummaryWorkDetail[];
}

export interface SummaryWorkDetail{
    date: Date;
    workShiftMinutes: number;
    workRecordMinutes: number;
    workShift?: WorkShift
}
